### PR TITLE
Declarative path

### PR DESCRIPTION
**Pull request by ghost**
Complex paths can be handled with Region sensors.  Given an ordered list of region sensors, when an entity enters a region, the motion of the entity is updated to navigate towards the next region in the list.

### CODE REVIEW

File: src/gameengine/collisiondetection/tree/Quad.java [hunk 1]
```diff
@@ -260,8 +269,7 @@ public void childEntityUpdated(double timeToCheck, Entity entity) {
 
     @Override
     public void relocateAndCheck(double timeToCheck, Entity entity) {
-        assert !isEntityInTree(entity) : "Entity should not be in the this tree when this method " +
-                "" + "is called";
+        assert !isEntityInTree(entity) : "Entity should not be in the this tree";
```
Comment (davidrusu) [inline src/gameengine/collisiondetection/tree/Quad.java]: `in the this` -> `in this`

File: src/Utilities/GameUtils.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
+package Utilities;
+
+import bricklets.Wall;
```
Comment (davidrusu) [inline src/Utilities/GameUtils.java:3]: tight coupling with the bricklets package is not so nice, maybe move the wall class to `entities/common/` ?

File: src/Utilities/LongQueue.java [hunk 3]
```diff
@@ -32,6 +32,8 @@ public LongQueue() {
      * @param initialCapacity The minimum initial capacity
      */
     public LongQueue(int initialCapacity) {
+        assert initialCapacity > 0;
```
Comment (davidrusu) [inline src/Utilities/LongQueue.java:35]: 👍

File: src/gameengine/entities/Entity.java [hunk 4]
```diff
@@ -21,7 +21,7 @@
     private int entityType = EntityType.STANDARD.ordinal();
     private int entityTypeBitMask = 1 << entityType;
     private Motion motion;
-    private Shape shape;
+    protected Shape shape;
```
Comment (davidrusu) [inline src/gameengine/entities/Entity.java]: why?

**Review by davidrusu — APPROVED** [review]
Looks good besides a few minor comments

File: src/Utilities/GameUtils.java (hunk 2 above)
Comment (ghost) [reply src/Utilities/GameUtils.java:3]: Agreed.  I felt ugly submitting that portion.  I planned to make the entity class accept handlers for updates & rendering so you no longer need to extend entity for trivial things (or maybe never) and then I'll delete the Wall class.

File: src/gameengine/entities/Entity.java (hunk 4 above)
Comment (ghost) [reply src/gameengine/entities/Entity.java]: I was too lazy to think when creating the Wall class.  Fixed back to private thanks.

**Pull request merged**